tests/backup/backup_test.go

```diff
@@ -3013,16 +3248,30 @@ func CreateRestore(restoreName string, backupName string,
 			Uid:  bkpUid,
 		},
 	}
-	_, err = backupDriver.CreateRestore(ctx, createRestoreReq)
-	log.FailOnError(err, "Creating restore")
+	_, err := backupDriver.CreateRestore(ctx, createRestoreReq)
+	if err != nil {
+		return err
+	}
 	restoreInspectRequest := &api.RestoreInspectRequest{
 		Name:  restoreName,
 		OrgId: orgID,
 	}
-	time.Sleep(time.Minute * 3)
-	resp, err := Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
-	log.FailOnError(err, "Verifying restore")
-	dash.VerifyFatal(resp.GetRestore().GetStatus().Status, api.RestoreInfo_StatusInfo_PartialSuccess, "Verifying restore")
+
+	restoreSuccessCheck := func() (interface{}, bool, error) {
+		resp, err := Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
+		log.FailOnError(err, "Failed verifying restore for - %s", restoreName)
+		if resp.GetRestore().GetStatus().GetStatus() == api.RestoreInfo_StatusInfo_PartialSuccess || resp.GetRestore().GetStatus().GetStatus() == api.RestoreInfo_StatusInfo_Success {
```

Comment:
since you are using resp.GetRestore().GetStatus() mulitple time, can you put it in a varible name and resuse it?

Reply:
Done